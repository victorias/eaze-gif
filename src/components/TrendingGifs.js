// @flow

import React, { PureComponent } from 'react';
import styled from 'styled-components';

import TrendingGif from '../containers/TrendingGif';

type Props = {
  onEnter: () => void,
  gifIds: Array<string>,
  isLoading: boolean,
};

const Container = styled.div`
  flex: 1 0 0;
  overflow-y: scroll;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  padding-left: 10vw;
  padding-right: 10vw;

  ${props =>
    props.isLoading &&
    `background-image: url('http://bestanimations.com/Science/Gears/loadinggears/loading-gear.gif');
    background-repeat: no-repeat;
    background-position: center center;`};
`;

const GifContainer = styled.div`
  flex: 0 0 auto;
  display: flex;
  margin: 0.5vw;

  @media (max-width: 480px) {
    margin: 0.5vw auto;
  }
`;

class TrendingGifs extends PureComponent<Props> {
  componentWillMount() {
    const { props } = this;
    props.onEnter();
  }

  render() {
    const { props } = this;

    if (props.isLoading || props.gifIds.length === 0) {
      return <Container isLoading={true} />;
    }

    return (
      <Container>
        {props.gifIds.map(id => (
          <GifContainer key={id}>
            <TrendingGif id={id} />
          </GifContainer>
        ))}
      </Container>
    );
  }
}

export default TrendingGifs;
