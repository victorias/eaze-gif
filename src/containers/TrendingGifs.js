// @flow

import { connect } from 'react-redux';

import TrendingGifs from '../components/TrendingGifs';
import * as actions from '../redux/gifs/actions';
import * as selectors from '../redux/gifs/selectors';

const mapStateToProps = state => ({
  gifIds: selectors.getTrendingList(state),
  isLoading: selectors.isLoading(state),
});

const mapDispatchToProps = {
  onEnter: actions.getTrending,
};

export default connect(mapStateToProps, mapDispatchToProps)(TrendingGifs);
