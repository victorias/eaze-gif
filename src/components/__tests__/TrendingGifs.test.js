// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import TrendingGifs from '../TrendingGifs';

describe('TrendingGifs', () => {
  it('renders loading', () => {
    const tree = renderer.create(
      <TrendingGifs isLoading gifIds={[]} onEnter={() => undefined} />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders an empty gif list', () => {
    const tree = renderer.create(
      <TrendingGifs isLoading={false} gifIds={[]} onEnter={() => undefined} />
    );
    expect(tree).toMatchSnapshot();
  });
});
