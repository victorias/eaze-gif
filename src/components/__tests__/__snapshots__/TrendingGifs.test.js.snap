// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingGifs renders an empty gif list 1`] = `
<div
  className="sc-EHOje bbZvAR"
/>
`;

exports[`TrendingGifs renders loading 1`] = `
<div
  className="sc-EHOje bbZvAR"
/>
`;
